packages/common/container-definitions/src/deltas.ts

```diff
@@ -61,21 +61,94 @@ export interface IDeltaSender {
     flush(): void;
 }
 
-/** Events emitted by the Delta Manager */
+/**
+ * Events emitted by {@link IDeltaManager}.
+ */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IDeltaManagerEvents extends IEvent {
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `messageBuffer`: TODO
+     */
     (event: "prepareSend", listener: (messageBuffer: any[]) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: TODO
+     */
     (event: "submitOp", listener: (message: IDocumentMessage) => void);
+
+    /**
+     * Emitted immediately after processing an incoming operation (op).
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: The op that was processed.
+     *
+     * - `processingTime`: TODO
+     *
+     * @internal
```

Comment:
It's hard for me to reason what @internal message conveys here. On one hand, I'd rather see it not used by anybody, and that likely deserves deprecated mark up. On another hand deprecation requires ensuring we understand current scenarios where it's used and having guidance on how to move away from it.
I think the most important note (here and below) - if it's used by container runtime, it can't be marked internal.
In a sense that we encourage (though do not make it easy) for any layer to be swappable. So, anybody should be able to re-implement IRuntime interface. And if we need it (for our own goodness), then we should assume someone else will have same need.

Reply:
Interesting. Okay. @ChumpChief you had noted that we should discourage external usage of this. If I'm understanding the situation correctly, this seems to be a contention between our *current* suggested consumption patterns and forward-looking extensibility considerations. I wonder what the best way to message this would be then. Or should we just omit any warnings and lean on ease of external discoverability (or lack thereof at this layer) to be our discouragement mechanism?

I'll remove the `@internal` tags for now, but this seems like an interesting category of issue to discuss further.

Reply:
I think Vlad and I are saying the same thing - that we don't want to see anybody use it.

Even for container runtime, it should probably be implemented differently (looks like we are just using it to watch for catchup, which probably has many alternate approaches we could consider).

Whatever tagging we add or not, I think it would be good to have some language here discouraging its use for general scenarios -- maybe in a `@remarks` section?  Something like "This event is not intended for general use.  Prefer to listen to events on the appropriate ultimate recipients of the ops, rather than listening to the ops directly on the IDeltaManager"